### DISCUSSION TITLE
I want the whole word to be found when searching for a part of a word How to do it ???

### DISCUSSION THREAD

**Discussion by Dmit595987** [Jeroen-G/Explorer#247] [Q&A] [answered]
I really like your package 

Here's mine explorer.php
    I make indexes simply through the model
```
`'default_index_settings' => [
        'index' => [
            'max_result_window' => 100000,
        ],
    ],
        'indexes' => [
        \App\Models\Course::class, ],
    `
```

next, I want to search
for "mas" and find "master" 
But it doesn't work at all

Course.php 
```
`    public function mappableAs(): array
    {
        return [
            'name' => [
                'type' => 'text',
                'analyzer' => 'ngram',
            ],
            'short_description' => 'text',
            'full_description' => 'text',
            'master_id' => 'keyword',
            'categories' => 'nested'
        ];
    }
    
        public function indexSettings(): array
    {

        $synonymAnalyzer = new StandardAnalyzer('ngram');
        $synonymAnalyzer->setFilters(['lowercase']);

        return (new Analysis())
            ->addAnalyzer($synonymAnalyzer)
            ->build();
    }`
```
    
    Help me please
    What am I doing wrong ????

**Comment by Jeroen-G** [comment]
What is the error and what exactly is not working? Do you have steps to reproduce the error? :)

**Reply by Dmit595987** [reply, marked as answer]
``Good afternoon! 
I have no mistakes. 
That's the problem. 

I'm looking for the word 'mas' through the library
And I expect it to find the word 'master' 

But it finds nothing

now i change explorer.php
`    'indexes' => [
            'courses' => [
                'model' => \App\Models\Course::class,
                'properties' => [
                    'name' => [
                        'type' => 'string',
                        'analyzer' => 'ngram_analyzer',
                        'search_analyzer' => 'standard',
                    ],
                ],
            ],
        ],
        'analyzers' => [
            'ngram_analyzer' => [
                'type' => 'custom',
                'tokenizer' => 'ngram_tokenizer',
                'filter' => ['lowercase'],
            ],
        ],
        'tokenizers' => [
            'ngram_tokenizer' => [
                'type' => 'ngram',
                'min_gram' => 2,
                'max_gram' => 20,
            ],`
